sklearnex/preview/ensemble/_forest.py

```diff
@@ -254,16 +421,30 @@ def __init__(
             max_samples=max_samples,
         )
 
-        # The splitter is recognized here for proper dispatching.
-        self._estimator = estimator
-        if self._estimator.__class__ == DecisionTreeClassifier:
-            self._onedal_classifier = onedal_RandomForestClassifier
-        elif self._estimator.__class__ == ExtraTreeClassifier:
-            self._onedal_classifier = onedal_ExtraTreesClassifier
-        else:
-            raise TypeError(
-                f"{estimator.__class__.__name__} is not a supported tree classifier"
-            )
+        # The estimator is checked against the class attribute for conformance.
```

Comment:
this is done to solve check_no_attributes_set_in_init tests/test_common.py test, but still allow for variation in the splitter by the user